aten/src/ATen/native/cpu/UpSampleKernelAVXAntialias.h

```diff
@@ -92,40 +92,48 @@ void pack_rgb(
 }
 
 void ImagingResampleHorizontalConvolution8u4x(
-    uint32_t* C10_RESTRICT lineOut0,
-    uint32_t* C10_RESTRICT lineOut1,
-    uint32_t* C10_RESTRICT lineOut2,
-    uint32_t* C10_RESTRICT lineOut3,
+    uint8_t* C10_RESTRICT lineOut0,
+    uint8_t* C10_RESTRICT lineOut1,
+    uint8_t* C10_RESTRICT lineOut2,
+    uint8_t* C10_RESTRICT lineOut3,
     int64_t out_xsize,
-    const uint32_t* C10_RESTRICT lineIn0,
-    const uint32_t* C10_RESTRICT lineIn1,
-    const uint32_t* C10_RESTRICT lineIn2,
-    const uint32_t* C10_RESTRICT lineIn3,
+    const uint8_t* C10_RESTRICT lineIn0,
+    const uint8_t* C10_RESTRICT lineIn1,
+    const uint8_t* C10_RESTRICT lineIn2,
+    const uint8_t* C10_RESTRICT lineIn3,
+    int64_t in_xsize,
     const int64_t* idx_ptr_xmin,
     const int64_t* idx_ptr_size,
     const int16_t* kk,
     int kmax,
-    unsigned int coefs_precision);
+    unsigned int coefs_precision,
+    int64_t num_channels,
+    bool is_last_line);
 
 void ImagingResampleHorizontalConvolution8u(
-    uint32_t* C10_RESTRICT lineOut,
+    uint8_t* C10_RESTRICT lineOut,
     int64_t out_xsize,
-    const uint32_t* C10_RESTRICT lineIn,
+    const uint8_t* C10_RESTRICT lineIn,
+    int64_t in_xsize,
     const int64_t* idx_ptr_xmin,
     const int64_t* idx_ptr_size,
     const int16_t* kk,
     int kmax,
-    unsigned int coefs_precision);
+    unsigned int coefs_precision,
+    int64_t num_channels,
+    bool is_last_line);
 
 void ImagingResampleVerticalConvolution8u(
-    uint32_t* C10_RESTRICT lineOut,
-    const uint32_t* C10_RESTRICT lineIn,
+    uint8_t* C10_RESTRICT lineOut,
+    const uint8_t* C10_RESTRICT lineIn,
     int64_t xsize,
     int64_t ids_min,
     int64_t ids_size,
     const int16_t* k,
-    unsigned int coefs_precision);
+    unsigned int coefs_precision,
+    int64_t num_channels);
 
+template<int num_channels>
```

Comment:
What's the reason for templating over `num_channels` here, instead of just using a normal parameter?

Reply:
With templated `num_channels` benchmarks are better by ~15us than without:
- with template:
```
PIL version:  9.0.0.post1
[------------------------------------------------------------ Resize -----------------------------------------------------------]
                                                                        |  Pillow (9.0.0.post1)  |  torch (2.1.0a0+gitcc42a3f) PR
1 threads: ----------------------------------------------------------------------------------------------------------------------
      3 torch.uint8 channels_last bilinear 256 -> (224, 224) aa=True    |         128.9          |              153.4
      4 torch.uint8 channels_last bilinear 256 -> (224, 224) aa=True    |                        |              144.3
      3 torch.uint8 channels_last bilinear 256 -> (224, 224) aa=False   |                        |              141.8
      4 torch.uint8 channels_last bilinear 256 -> (224, 224) aa=False   |                        |              131.4
      3 torch.uint8 channels_last bilinear 256 -> (256, 224) aa=True    |          91.1          |              113.9
      4 torch.uint8 channels_last bilinear 256 -> (256, 224) aa=True    |                        |              102.9
      3 torch.uint8 channels_last bilinear 256 -> (256, 224) aa=False   |                        |              106.1
      4 torch.uint8 channels_last bilinear 256 -> (256, 224) aa=False   |                        |               94.0
      3 torch.uint8 channels_last bilinear 256 -> (224, 256) aa=True    |          52.4          |               52.8
      4 torch.uint8 channels_last bilinear 256 -> (224, 256) aa=True    |                        |               55.4
      3 torch.uint8 channels_last bilinear 256 -> (224, 256) aa=False   |                        |               48.5
      4 torch.uint8 channels_last bilinear 256 -> (224, 256) aa=False   |                        |               49.8
      3 torch.uint8 channels_first bilinear 256 -> (224, 224) aa=True   |         127.8          |              296.8
      4 torch.uint8 channels_first bilinear 256 -> (224, 224) aa=True   |                        |              251.4
      3 torch.uint8 channels_first bilinear 256 -> (224, 224) aa=False  |                        |              285.5
      4 torch.uint8 channels_first bilinear 256 -> (224, 224) aa=False  |                        |              239.6
      3 torch.uint8 channels_first bilinear 256 -> (256, 224) aa=True   |          91.8          |              269.8
      4 torch.uint8 channels_first bilinear 256 -> (256, 224) aa=True   |                        |              228.4
      3 torch.uint8 channels_first bilinear 256 -> (256, 224) aa=False  |                        |              261.1
      4 torch.uint8 channels_first bilinear 256 -> (256, 224) aa=False  |                        |              222.6
      3 torch.uint8 channels_first bilinear 256 -> (224, 256) aa=True   |          52.6          |              220.5
      4 torch.uint8 channels_first bilinear 256 -> (224, 256) aa=True   |                        |              183.5
      3 torch.uint8 channels_first bilinear 256 -> (224, 256) aa=False  |                        |              216.1
      4 torch.uint8 channels_first bilinear 256 -> (224, 256) aa=False  |                        |              178.7

Times are in microseconds (us).
```
- without template:
```
PIL version:  9.0.0.post1
[------------------------------------------------------------ Resize -----------------------------------------------------------]
                                                                        |  Pillow (9.0.0.post1)  |  torch (2.1.0a0+gitcc42a3f) PR
1 threads: ----------------------------------------------------------------------------------------------------------------------
      3 torch.uint8 channels_last bilinear 256 -> (224, 224) aa=True    |         128.8          |              161.7
      4 torch.uint8 channels_last bilinear 256 -> (224, 224) aa=True    |                        |              158.6
      3 torch.uint8 channels_last bilinear 256 -> (224, 224) aa=False   |                        |              150.6
      4 torch.uint8 channels_last bilinear 256 -> (224, 224) aa=False   |                        |              145.8
      3 torch.uint8 channels_last bilinear 256 -> (256, 224) aa=True    |          91.7          |              122.1
      4 torch.uint8 channels_last bilinear 256 -> (256, 224) aa=True    |                        |              116.5
      3 torch.uint8 channels_last bilinear 256 -> (256, 224) aa=False   |                        |              113.6
      4 torch.uint8 channels_last bilinear 256 -> (256, 224) aa=False   |                        |              107.7
      3 torch.uint8 channels_last bilinear 256 -> (224, 256) aa=True    |          52.7          |               53.7
      4 torch.uint8 channels_last bilinear 256 -> (224, 256) aa=True    |                        |               56.1
      3 torch.uint8 channels_last bilinear 256 -> (224, 256) aa=False   |                        |               49.8
      4 torch.uint8 channels_last bilinear 256 -> (224, 256) aa=False   |                        |               52.1
      3 torch.uint8 channels_first bilinear 256 -> (224, 224) aa=True   |         128.8          |              312.7
      4 torch.uint8 channels_first bilinear 256 -> (224, 224) aa=True   |                        |              267.0
      3 torch.uint8 channels_first bilinear 256 -> (224, 224) aa=False  |                        |              300.2
      4 torch.uint8 channels_first bilinear 256 -> (224, 224) aa=False  |                        |              254.9
      3 torch.uint8 channels_first bilinear 256 -> (256, 224) aa=True   |          91.8          |              283.8
      4 torch.uint8 channels_first bilinear 256 -> (256, 224) aa=True   |                        |              246.5
      3 torch.uint8 channels_first bilinear 256 -> (256, 224) aa=False  |                        |              277.1
      4 torch.uint8 channels_first bilinear 256 -> (256, 224) aa=False  |                        |              238.0
      3 torch.uint8 channels_first bilinear 256 -> (224, 256) aa=True   |          53.3          |              225.8
      4 torch.uint8 channels_first bilinear 256 -> (224, 256) aa=True   |                        |              187.4
      3 torch.uint8 channels_first bilinear 256 -> (224, 256) aa=False  |                        |              219.3
      4 torch.uint8 channels_first bilinear 256 -> (224, 256) aa=False  |                        |              181.6

Times are in microseconds (us).
```